lib/browser/api/menu-item.js

```diff
@@ -7,18 +7,8 @@ let nextCommandId = 0
 const MenuItem = function (options) {
   const {Menu} = require('electron')
 
-  this.selector = options.selector
-  this.type = options.type
-  this.role = options.role
-  this.label = options.label
-  this.sublabel = options.sublabel
-  this.accelerator = options.accelerator
-  this.icon = options.icon
-  this.enabled = options.enabled
-  this.visible = options.visible
-  this.checked = options.checked
-
-  this.submenu = options.submenu
+  Object.assign(this, options)
```

Comment:
@zeke What I propose is adding an "undesired" array of properties

```javascript
var undesiredProperties = ['overrideProperty', 'overrideReadOnlyProperty' /* add more if necessary*/]
```

and just before calling `.assign`, `delete` the properties from `options` which are in that array, aka
```javascript
undesiredProperties.forEach((undesiredProperty) => delete options[undesiredProperty])
```

Otherwise, an alternative would to either make methods like `overrideProperty` unwritable, or possibly put these customProperties inside another property like "custom" so passing properties like `overrideProperty` will not impact the existing method.

```javascript
this.custom = Object.assign({}, getCustomProps(options))
```

In all cases, it'd be important to stay consistent between the handling of customProperties when using `buildFromTemplate` or through `append`